proxy/hdrs/HTTP.h

```diff
@@ -700,29 +696,42 @@ class HTTPHdr : public MIMEHdr
 /*-------------------------------------------------------------------------
   -------------------------------------------------------------------------*/
 
-inline HTTPVersion::HTTPVersion(int32_t version) : m_version(version) {}
+inline HTTPVersion::HTTPVersion(int version)
+{
+  vmajor = (version & 0xFFFF0000) >> 16;
```

Comment:
Better as `vmajor = (version >> 16) & 0xFFFF;`

Reply:
Why?

Reply:
It looks more consistent with the next line, and avoids any issues with the sign bit.